### PR TITLE
Updated Documentation on StartGuiManagerScript.cs and Manager.cs

### PR DESCRIPTION
**Pull request by nsmonagh**
- Committed recommendations to documentation on StartGuiManagerScript.cs
- Performed a cleanup of syntax on Manager.cs, mainly null propagation.
- Closed issue #184 (though a rather minor one)
- Added around 170 lines of documentation to Manager.cs

Note: I have decided that I will be documenting the Manager.cs file in its entirety. Since I have only thoroughly learned how the functions in StartGuiManagerScript.cs work, there may be inaccuracies in my summaries or they may be badly worded.
If my work is being wasted trying to document everything, I will narrow my focus to functions that pertain directly to StartGuiManagerScript.cs

Again feedback is always welcome.

### CODE REVIEW

**Comment by nsmonagh** [comment]
Also I was uncertain what exactly the Manager.GetUVFromMapCoordinates() function does as well as the Manager.RegisterUndoStackUpdateOp(System.Action op) and similar functions do.

File: Assets/Scripts/WorldEngine/Manager.cs [hunk 1]
```diff
@@ -285,93 +282,32 @@ public class Debug_TracingData
...
     public XmlAttributeOverrides AttributeOverrides { get; private set; }
 
-    public static bool PerformingAsyncTask
-    {
-        get
-        {
-            return _manager._performingAsyncTask;
-        }
-    }
+    public static bool PerformingAsyncTask => _manager._performingAsyncTask;
```
Comment (Zematus) [inline Assets/Scripts/WorldEngine/Manager.cs]: Another thing I didn't know was possible. nice! BTW. What IDE/editor are you using?

File: Assets/Scripts/WorldEngine/Manager.cs [hunk 2]
```diff
@@ -386,11 +322,17 @@ private Manager()
 
         _lastUpdatedCells = new HashSet<TerrainCell>();
 
-        /// static initalizations
+        // static initializations
 
         Tribe.GenerateTribeNounVariations();
     }
 
+    /// <summary>Determines if the required control and shift keys were used.</summary>
```
Comment (Zematus) [inline Assets/Scripts/WorldEngine/Manager.cs]: This function also checks that user is not interacting with an Input Field in the UI before allowing the key shortcut to proceed

**Comment by Zematus** [comment]
> Also I was uncertain what exactly the Manager.GetUVFromMapCoordinates() function does as well as the Manager.RegisterUndoStackUpdateOp(System.Action op) and similar functions do.

GetUVFromMapCoordinates() converts world map coordinates (longitude, latitude) to a texture's UV coordinates, which go from 0 to 1 on both x and y coordinates, (0,0) being the bottom left corner and (1,1) the top right corner, IRC. This is what I use to map cell events to points in the texture, which then are mapped to the 2D map or 3D sphere surfaces. I use this for example to re-center the map when the player clicks the "goto" button on an cell event dialog. 

RegisterUndoStackUpdateOp() (and the others) are rather simple, they register or deregister listeners to the events related to adding or removing Undo/Redo actions to/from their respective stacks. I use them to notify the editor toolbar when to enable/disable the Undo and Redo buttons in accordance to those stack states. For example, if no undo action is on the stack, then the undo button should appear disabled since there's nothing that can be undone. But if one or more actions are pushed into the stack then the button should appear enabled. In theory, other event listeners could be registered with those functions but so far I only use them to control the toolbar button states.

**Comment by Zematus** [comment]
Let me know when this PR is ready to merge and I'll proceed. Otherwise you can keep adding more changes to it.

File: Assets/Scripts/WorldEngine/Manager.cs (hunk 1 above)
Comment (nsmonagh) [reply Assets/Scripts/WorldEngine/Manager.cs]: I'm using visual studio 2019

**Comment by nsmonagh** [comment]
I probably won't be done documenting the Manager.cs file for at least a couple more days. But when I am done, I'll update my PR and notify you again.

**Pull request closed without merging**